src/main/java/org/manifold/compiler/front/ExpressionGraph.java

```diff
@@ -201,8 +209,12 @@ public void addSubExpressionGraph(ExpressionGraph subGraph,
 
     // each edge in subgraph -> edge in main graph should refer to the same source/target
     subGraph.getEdges().forEach(edge -> {
-        exprEdgeMap.get(edge).setSource(exprVertexMap.get(edge.getSource()));
-        exprEdgeMap.get(edge).setTarget(exprVertexMap.get(edge.getTarget()));
+        if (edge.getSource() != null) {
```

Comment:
There's another if statement after the end of this if statement, so that wouldn't work.